common/src/types/tasks/descriptors/mod.rs

```diff
@@ -0,0 +1,297 @@
+                unimplemented!("SettleMatchInternal should preempt queue, no key needed")
+            },
+            TaskDescriptor::UpdateMerkleProof(task) => task.wallet.wallet_id,
+            TaskDescriptor::UpdateWallet(task) => task.wallet_id,
+            TaskDescriptor::NodeStartup(task) => task.id,
+        }
+    }
+
+    /// Returns whether the task is a wallet task
+    ///
+    /// Currently all tasks are wallet tasks
+    pub fn is_wallet_task(&self) -> bool {
+        match self {
+            TaskDescriptor::NewWallet(_)
+            | TaskDescriptor::LookupWallet(_)
+            | TaskDescriptor::OfflineFee(_)
+            | TaskDescriptor::RelayerFee(_)
+            | TaskDescriptor::RedeemRelayerFee(_)
+            | TaskDescriptor::UpdateWallet(_)
+            | TaskDescriptor::SettleMatch(_)
+            | TaskDescriptor::SettleMatchInternal(_)
+            | TaskDescriptor::UpdateMerkleProof(_) => true,
+            TaskDescriptor::NodeStartup(_) => false,
+        }
+    }
+
+    /// Get a human readable description of the task
+    pub fn display_description(&self) -> String {
+        match self {
+            TaskDescriptor::NewWallet(_) => "New Wallet".to_string(),
+            TaskDescriptor::UpdateWallet(args) => args.description.display_description(),
+            TaskDescriptor::LookupWallet(_) => "Lookup Wallet".to_string(),
+            TaskDescriptor::SettleMatch(_) => "Settle Match".to_string(),
+            TaskDescriptor::SettleMatchInternal(_) => "Settle Match".to_string(),
+            TaskDescriptor::OfflineFee(_) => "Pay Fee Offline".to_string(),
+            TaskDescriptor::RelayerFee(_) => "Pay Relayer Fee".to_string(),
+            TaskDescriptor::RedeemRelayerFee(_) => "Redeem Relayer Fee".to_string(),
+            TaskDescriptor::UpdateMerkleProof(_) => "Update Merkle Proof".to_string(),
+            TaskDescriptor::NodeStartup(_) => "Node Startup".to_string(),
+        }
+    }
+}
+
+// ---------
+// | Mocks |
+// ---------
+
+#[cfg(any(test, feature = "mocks"))]
+pub mod mocks {
+    //! Mocks for the task descriptors
+    use circuit_types::keychain::SecretSigningKey;
+    use ethers::core::utils::keccak256;
+    use ethers::signers::Wallet as EthersWallet;
+    use k256::ecdsa::SigningKey as K256SigningKey;
+    use util::get_current_time_millis;
+
+    use crate::types::{
+        gossip::mocks::mock_peer, tasks::TaskIdentifier, wallet::Wallet,
+        wallet_mocks::mock_empty_wallet,
+    };
+
+    use super::{
+        NewWalletTaskDescriptor, QueuedTask, QueuedTaskState, TaskDescriptor, TaskQueueKey,
+    };
+
+    /// Generate the wallet update signature for a new wallet
+    pub fn gen_wallet_update_sig(wallet: &Wallet, key: &SecretSigningKey) -> Vec<u8> {
+        // Serialize and hash the wallet commitment
+        let new_wallet_comm = wallet.get_wallet_share_commitment();
+        let digest = keccak256(new_wallet_comm.to_biguint().to_bytes_be());
```

Comment:
pls no, we should be importing `contracts_common::custom_serde::BytesSerializable` and doing:
```suggestion
        let digest = keccak256(new_wallet_comm.inner().serialize_to_bytes());
```

Reply:
done, pulled in changes from your PR on main